rfcs/20220307-monitoring_across_the_stack.md

```diff
@@ -0,0 +1,353 @@
+
+### Other extremely low level monitoring tools
+
+There are also some very low level [Linux daemon based time-series databases](https://github.com/oetiker/rrdtool-1.x) that are quite capable and can integrate with Grafana. However, they are all very low level and mostly do not support all of the three features I listed above.
+
+### DataDog
+
+[DataDog](https://www.datadoghq.com/) is a SaaS offerring that runs an open source stack mixed with some proprietary magic. Prometheus and Grafana together very much resemble DataDog, even down to the UI.
+
+DataDog is fantastic. But it also comes at a higher cost than self-hosted Prometheus and Grafana. It is also ultimately a closed system. They provide ways for getting your data out and the client libraries for sending events used are the same (or very similar) to the ones for Graphite and Prometheus but it would still put work on alternative Openverse instances to make significant changes to the monitoring backend.
+
+DataDog does have very good documentation and holds your hand a _lot_ when it comes to setting up monitors and dashboards. If I had to choose a hosted proprietary solution I'd go with DataDog over AWS. However, I would not choose it over hosted Prometheus and Grafana if given the choice.
+
+### Solar Winds stack
+
+Solar Winds also has their own [server and application monitoring tool](https://www.solarwinds.com/server-application-monitor/use-cases). It's meant to be easy to deploy and automatically configures itself for the services you have running. I don't have any experience with this other than having used it to monitor Tomcat services at a previous job. My impression is that it's targeted mostly at large enterprise and government contracts rather than projects like ours. The services themselves are also not free software, though the SDKs that integrate with them are. Likewise, it doesn't seem to be a very flexible offering; it would probably be difficult to integrate things like monitoring a Nuxt service.
+
+### The ELK stack
+
+The final alternative is the [ELK stack](https://www.elastic.co/what-is/elk-stack). It's not exactly a 1:1 alternative to Prometheus and Grafana, though it could fulfill [some](https://www.elastic.co/guide/en/kibana/current/xpack-ml-anomalies.html) of the [same](https://www.elastic.co/kibana/) use cases. This is a strong alternative that I hope reviewers of this RFC spend time considering. Read through Elastic's documentation about it and see if it could make sense for us. It could also be something that we implement down the road as a supplement to Prometheus and Grafana. Or it can replace it altogether.
```

Comment:
I cannot say I have a ton of experience with the ELK stack, but what I remember wasn't too enjoyable. Kibana was difficult to grok and navigate. I remember our ops team at a previous job having lots of trouble setting it all up. That said, all of this is tertiary and I don't have any specific, data-backed resistance to it.

Reply:
I don't think Kibana's queries are all that much more cryptic than Prometheus, FWIW. Prometheus might be even more so, it's extremely syntax heavy whereas Kibana has a tendency to be pretty verbose.